src/components/LayoutTree.tsx

```diff
@@ -707,6 +706,9 @@ const LayoutTree = () => {
                   // Do this as padding instead of y, otherwise there will be a gap between drop targets.
                   style={cliff < 0 ? cliffPaddingStyle : undefined}
                   crossContextualKey={key}
+                  cliff={cliff}
+                  index={index}
+                  treeThoughts={treeThoughtsPositioned}
```

Comment:
VirtualThought should definitely not need to know about the entire list of `treeThoughtsPositioned`. That breaks the hierarchy and encapsulation we've established. You're going to have to find a way to accomplish this that respects the division of responsibility.

Reply:
Then I think we can simply take it to `LayoutTree` and pass it from there like this, 

```
const prevThought = treeThoughtsPositioned[index - 1]
const prevCliff = prevThought ? prevThought.cliff : undefined
```